### PR TITLE
Added Tests For iOS Text Embedder

### PR DESCRIPTION
**Pull request by priankakariatyml**
1. Added Cosine Similarity Api
2. Added objective c and swift tests

### CODE REVIEW

File: mediapipe/tasks/ios/test/text/text_embedder/MPPTextEmbedderTests.m [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  MPPEmbedding *embedding2 = [self assertFloatEmbeddingResultsOfEmbedText:kText2
+                                                        usingTextEmbedder:textEmbedder
+                                                                 hasCount:16
+                                                               firstValue:0.0312863f];
+
+  NSNumber *cosineSimilarity = [MPPTextEmbedder cosineSimilarityBetweenEmbedding1:embedding1
+                                                                    andEmbedding2:embedding2
+                                                                            error:nil];
+  XCTAssertEqualWithAccuracy(cosineSimilarity.doubleValue, 0.999937f, kDoubleDiffTolerance);
+}
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/ios/test/text/text_embedder/MPPTextEmbedderTests.m:140]: Can you also port the newly added test "SucceedsWithMobileBertAndDifferentThemes"?

File: mediapipe/tasks/ios/test/text/text_embedder/TextEmbedderTests.swift [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    assertEmbeddingIsFloat(textEmbedderResult.embeddingResult.embeddings[0])
+    assertEmbedding(
+      textEmbedderResult.embeddingResult.embeddings[0].floatEmbedding!,
+      hasCount: embeddingCount,
+      hasFirstValue: firstValue)
+
+    return textEmbedderResult.embeddingResult.embeddings[0]
+  }
+
+  func testEmbedWithBertSucceeds() throws {
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/ios/test/text/text_embedder/TextEmbedderTests.swift:88]: I think we should add a comment to this file stating that this is deliberately a subset of tests and that the canonical tests are the ObjC ones - or we should port all the tests. I don't want someone to think that we have poor test coverage for our iOS library.

File: mediapipe/tasks/ios/text/text_embedder/sources/MPPTextEmbedder.h [hunk 3]
```diff
@@ -86,6 +86,23 @@ NS_SWIFT_NAME(TextEmbedder)
 
 - (instancetype)init NS_UNAVAILABLE;
 
+/** Utility function to compute[cosine similarity](https://en.wikipedia.org/wiki/Cosine_similarity)
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/ios/text/text_embedder/sources/MPPTextEmbedder.h:89]: Please rewrap.

File: mediapipe/tasks/ios/text/text_embedder/sources/MPPTextEmbedder.h [hunk 4]
```diff
@@ -86,6 +86,23 @@ NS_SWIFT_NAME(TextEmbedder)
...
+ * computed.
+ * @param error An optional error parameter populated when there is an error in calculating cosine
+ * similarity between two embeddings.
+ *
+ * @return An `NSNumber` which holds the cosine similarity of type `double`.
+ */
++ (nullable NSNumber *)cosineSimilarityBetweenEmbedding1:(MPPEmbedding *)embedding1
+                                           andEmbedding2:(MPPEmbedding *)embedding2
+                                                   error:(NSError **)error NS_SWIFT_NAME(cosineSimilarity(embedding1:embedding2:));
+    // NS_SWIFT_NAME(cosineSimilarity(embedding1: embedding2:));
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/ios/text/text_embedder/sources/MPPTextEmbedder.h:104]: Remove?

File: mediapipe/tasks/ios/test/text/text_embedder/MPPTextEmbedderTests.m (hunk 1 above)
Comment (priankakariatyml) [reply mediapipe/tasks/ios/test/text/text_embedder/MPPTextEmbedderTests.m:242]: Fixed. Added a test for quantised embedding as well.

File: mediapipe/tasks/ios/test/text/text_embedder/TextEmbedderTests.swift (hunk 2 above)
Comment (priankakariatyml) [reply mediapipe/tasks/ios/test/text/text_embedder/TextEmbedderTests.swift:93]: I have added a note for now. We can duplicate all tests in Swift if that is more appropriate. I am just following our approach in the tflite-support library for the time being.

**Review by schmidt-sebastian — APPROVED** [review]

**Pull request merged**